packages/core/database/src/migrations/index.ts

```diff
@@ -70,11 +70,7 @@ const createUmzugProvider = (db: Database) => {
 
 // NOTE: when needed => add internal migrations for core & plugins. How do we overlap them with users migrations ?
 
-/**
- * Creates migrations provider
- * @type {import('.').createMigrationsProvider}
- */
-export const createMigrationsProvider = (db: Database): MigrationProvider => {
+export const createMigrationsProvider = async (db: Database): Promise<MigrationProvider> => {
```

Comment:
Can you remind me why this should be async ?  FYI we will have a conflict in v5 here

Reply:
this one needs to be async for [typescript migration compilation](https://github.com/strapi/strapi/pull/18010)

Reply:
Can we delay this for v5 as I refactor this whole part and we might need to redo the change 🤔

Reply:
Up to you, I'm happy to wait until v5 as long as we don't worry about not supporting typescript migrations in v4

Reply:
Yeah I think that's fine delaying it.